FileIO/TetGenInterface.cpp

```diff
@@ -230,8 +228,18 @@ MeshLib::Mesh* TetGenInterface::readTetGenMesh (std::string const& nodes_fname,
 		return nullptr;
 	}
 
+	MeshLib::Properties properties;
+	// remove material vector again if all mat-ids are "0"
+	if (std::any_of(materials.cbegin(), materials.cend(), [](int m){ return m != 0; }))
+	{
+		boost::optional<MeshLib::PropertyVector<int> &> mat_props = 
+			properties.createNewPropertyVector<int>("MaterialIDs", MeshLib::MeshItemType::Cell);
+		mat_props->resize(elements.size());
```

Comment:
same as in the other comment ... properties has just been created specifically for returning the mat vector....